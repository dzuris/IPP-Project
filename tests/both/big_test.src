.ippcode22

DEFVAR GF@temp

# Read six inputs
# 1
READ GF@temp int
PUSHS GF@temp

# 2
READ GF@temp int
PUSHS GF@temp

# 3
READ GF@temp int
PUSHS GF@temp

# 4
READ GF@temp int
PUSHS GF@temp

# 5
READ GF@temp int
PUSHS GF@temp

# 6
READ GF@temp int
PUSHS GF@temp

CREATEFRAME
DEFVAR TF@temp
POPS TF@temp

PUSHFRAME
CREATEFRAME
DEFVAR TF@temp
POPS TF@temp

ADD GF@temp TF@temp LF@temp
WRITE GF@temp
WRITE string@\010

JUMPIFEQ label1 GF@temp int@51

WRITE string@toto\032ma\032byt\032preskocene\010

LABEL label1

WRITE string@toto\032by\032sa\032malo\032vytlacit\010

POPS LF@temp
POPFRAME

POPS GF@temp

DEFVAR GF@new_var
MUL GF@new_var GF@temp TF@temp

WRITE GF@new_var
WRITE string@\010

POPS GF@temp
POPS GF@new_var

DEFVAR GF@cnt
MOVE GF@cnt GF@new_var

Label start
JUMPIFEQ end GF@cnt int@0

WRITE GF@temp
WRITE string@\032
SUB GF@temp GF@temp int@1
SUB GF@cnt GF@cnt int@1
JUMP start

LABEL end
WRITE string@\010
