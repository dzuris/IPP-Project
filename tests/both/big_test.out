51
toto by sa malo vytlacit
1020
5 4 3 2 1 0 -1 -2 -3 -4 
